pkgs/desktops/mate/libmatemixer/default.nix

```diff
@@ -1,6 +1,14 @@
-{ config, lib, stdenv, fetchurl, pkg-config, gettext, glib
-, alsaSupport ? stdenv.isLinux, alsa-lib
-, pulseaudioSupport ? config.pulseaudio or true, libpulseaudio
+{ config
+, lib
+, stdenv
+, fetchurl
+, pkg-config
+, gettext
+, glib
+, alsaSupport ? stdenv.isLinux
+, alsa-lib
+, pulseaudioSupport ? config.pulseaudio or true
```

Comment:
This should be stdenv.isLinux

Reply:
Why?

It seems that there is not a consensus on nixpkgs:
``` console
$ grep -r -i 'pulseaudiosupport ?'
pkgs/os-specific/linux/guvcview/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
pkgs/desktops/mate/mate-settings-daemon/default.nix:, pulseaudioSupport ? stdenv.config.pulseaudio or true
pkgs/desktops/mate/libmatemixer/default.nix:, pulseaudioSupport ? config.pulseaudio or true
pkgs/development/libraries/pcaudiolib/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
pkgs/development/libraries/ffmpeg/generic.nix:, pulseaudioSupport ? stdenv.isLinux
pkgs/development/libraries/SDL/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux && !stdenv.hostPlatform.isAndroid, libpulseaudio
pkgs/development/libraries/audio/rtaudio/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.hostPlatform.isLinux
pkgs/development/libraries/audio/roc-toolkit/default.nix:  pulseaudioSupport ? true,
pkgs/development/libraries/SDL2/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux && !stdenv.hostPlatform.isAndroid
pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:, pulseaudioSupport ? mediaSupport
pkgs/applications/networking/browsers/firefox/common.nix:, pulseaudioSupport ? stdenv.isLinux, libpulseaudio
pkgs/applications/networking/instant-messengers/zoom-us/default.nix:, pulseaudioSupport ? true
pkgs/applications/radio/gqrx/default.nix:, pulseaudioSupport ? true, libpulseaudio
pkgs/applications/emulators/wine/default.nix:  pulseaudioSupport ? false,
pkgs/applications/emulators/rpcs3/default.nix:, pulseaudioSupport ? true, libpulseaudio
pkgs/applications/video/obs-studio/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
pkgs/applications/video/aegisub/default.nix:, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
pkgs/applications/audio/cmus/default.nix:, pulseaudioSupport ? config.pulseaudio or false, libpulseaudio ? null
pkgs/applications/audio/reaper/default.nix:, pulseaudioSupport ? config.pulseaudio or true
pkgs/applications/audio/snapcast/default.nix:, pulseaudioSupport ? false, libpulseaudio
pkgs/applications/audio/audio-recorder/default.nix:, pulseaudioSupport ? true, libpulseaudio }:
pkgs/applications/audio/mimic/default.nix:, pulseaudioSupport ? config.pulseaudio or false, libpulseaudio }:
pkgs/tools/video/replay-sorcery/default.nix:, pulseaudioSupport ? true, libpulseaudio
```

Reply:
pulseaudio only works on linux. Also config.pulseaudio is mostly undocumented and not well defined.

Changing it to stdenv.isLinux makes darwin support easier.

Reply:
What is the meaning of `or` in this construction? I suppose it is the logical disjunction. I have found the function `or` in lib, but it is not a binary operator.

Are `stdenv.isLinux or false` and `stdenv.isLinux` equivalent?

Reply:
Never mind. I have found the documentation: https://nixos.org/manual/nix/stable/expressions/language-operators.html.

This `or` is not an operator on itself, but it is an optional part of the syntax of the select (`.`) operator.

Reply:
> pulseaudio only works on linux. Also config.pulseaudio is mostly undocumented and not well defined.
> 
> Changing it to stdenv.isLinux makes darwin support easier.

But it would be more restrictive, as the global configuration to enable/disable pulseaudio would be ignored.

Maybe `config.pulseaudio or stdenv.isLinux` is a better default.

Or even `config.pulseaudio or stdenv.isLinux or false`.

Reply:
> Are `stdenv.isLinux or false` and `stdenv.isLinux` equivalent?

yes. It is only required for config.pulseaudio because that is most of the times undefined.

Reply:
> as the global configuration to enable/disable pulseaudio would be ignored.

There is no global way to disable pulseaudio at compile time. There are many many ways and each package does it in their own way.

Reply:
I am still not sure what is better between `pulseaudioSupport ? config.pulseaudio or stdenv.isLinux or false` and `pulseaudioSupport ? stdenv.isLinux or false`. What is your final advice here?

Reply:
We don't need the ``or false`` since stdenv.isLinux is false on other platforms.

I would go with ``pulseaudioSupport ? stdenv.isLinux``